includes/options/views/class-amp-options-manager.php

```diff
@@ -27,58 +152,46 @@ public static function update_option( $option, $value ) {
 		return update_option( self::OPTION_NAME, $amp_options, false );
 	}
 
+	/**
+	 * Handle analytics submission.
+	 */
 	public static function handle_analytics_submit() {
-		// Request must come from user with right capabilities
+
+		// Request must come from user with right capabilities.
 		if ( ! current_user_can( 'manage_options' ) ) {
-			wp_die( __( 'Sorry, you do not have the necessary permissions to perform this action', 'amp' ) );
+			wp_die( esc_html__( 'Sorry, you do not have the necessary permissions to perform this action', 'amp' ) );
 		}
-		// Ensure request is coming from analytics option form
+
+		// Ensure request is coming from analytics option form.
 		check_admin_referer( 'analytics-options', 'analytics-options' );
 
-		$status = AMP_Options_Manager::update_analytics_options( $_POST );
+		if ( isset( $_POST['amp-options']['analytics'] ) ) {
+			self::update_option( 'analytics', wp_unslash( $_POST['amp-options']['analytics'] ) );
 
-		// Redirect to keep the user in the analytics options page
-		// Wrap in is_admin() to enable phpunit tests to exercise this code
-		wp_safe_redirect( admin_url( 'admin.php?page=amp-analytics-options&valid=' . $status ) );
-		exit;
-	}
-
-	public static function update_analytics_options( $data ) {
-		// Check save/delete pre-conditions and proceed if correct
-		if ( empty( $data['vendor-type'] ) || empty( $data['config'] ) ) {
-			return false;
-		}
-
-		// Validate JSON configuration
-		$is_valid_json = AMP_HTML_Utils::is_valid_json( stripslashes( $data['config'] ) );
-		if ( ! $is_valid_json ) {
-			return false;
-		}
-		$amp_analytics = self::get_option( 'analytics', array() );
-
-		$entry_vendor_type = sanitize_key( $data['vendor-type'] );
-		$entry_config = stripslashes( trim( $data['config'] ) );
-
-		if ( ! empty( $data['id-value'] ) ) {
-			$entry_id = sanitize_key( $data['id-value'] );
-		} else {
-			// Generate a hash string to uniquely identify this entry
-			$entry_id = substr( md5( $entry_vendor_type . $entry_config ), 0, 12 );
-			// Avoid duplicates
-			if ( isset( $amp_analytics[ $entry_id ] ) ) {
-				return false;
+			$errors = get_settings_errors( self::OPTION_NAME );
+			if ( empty( $errors ) ) {
+				add_settings_error( self::OPTION_NAME, 'settings_updated', __( 'The analytics entry was successfully saved!', 'amp' ), 'updated' );
+				$errors = get_settings_errors( self::OPTION_NAME );
 			}
+			set_transient( 'settings_errors', $errors );
 		}
 
-		if ( isset( $data['delete'] ) ) {
-			unset( $amp_analytics[ $entry_id ] );
-		} else {
-			$amp_analytics[ $entry_id ] = array(
-				'type' => $entry_vendor_type,
-				'config' => $entry_config,
-			);
-		}
+		// Redirect to keep the user in the analytics options page.
+		// Wrap in is_admin() to enable phpunit tests to exercise this code.
+		wp_safe_redirect( admin_url( 'admin.php?page=amp-analytics-options&settings-updated=1' ) );
```

Comment:
Yes, but this is how it is currently in the released version of the plugin: https://github.com/Automattic/amp-wp/blob/6df01e9c90f2de3c4cb16952401a285992d11021/includes/options/views/class-amp-options-manager.php#L42

I didn't want to spend more time on improving analytics settings in the admin since we want to move them to the Customizer.